content/en/docs/How Tos/metrics.md

```diff
@@ -134,13 +72,171 @@ or an error saying why it can't.
 
-Once a [`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient)
-instance has been successfully created, it's made available to and used by
-a Baker pipeline to report metrics. During construction, components receive the 
-[`MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) instance.
+```go
+func (f *MyFilter) Process(r Record, next func(Record)) {
+    atomic.AddInt64(&myFilter.totalLines, 1)
+
+    /* perform http request */
+    f.mu.Lock() // keep track of its duration
+    f.requestDurations = append(f.requestDurations, duration)
+    f.mu.Unlock()
+
+    if (/* filter logic*/) {
+        // discard line
+        atomic.AddInt64(&f.filteredLines, 1)
+        return
+    }
+}
+
+func (f *MyFilter) Stats() baker.FilterStats {
+    // copy the slice of durations
+    f.mu.Lock()
+    requestDurations := f.requestDurations[:]
+    f.mu.Unlock()
+
+    bag := make(baker.MetricsBag)
+    bag.AddTimings("myfilter_http_request_duration", requestDurations)
+    return baker.FilterStats{
+        NumProcessedLines: atomic.LoadInt64(&f.totalLines),
+        NumFilteredLines: atomic.LoadInt64(&f.filteredLines),
+        Metrics: bag,
+    }
+}
+```
+
+#### MetricsClient Example
 
+Once a 
 [`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) 
-supports the most common type of metric types: *gauges*, *counters* and *histograms*.
+instance has been successfully created, it's made available to and used by a Baker 
+pipeline to report metrics. During construction, components receive the 
+[`MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) instance.
+[`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) 
+provides a set of methods to report the most common type of metric types (e.g. 
+*gauges*, *counters* and *histograms*)
+
+Taking the previous example of the `MetricsBag`, we now consider the situation in 
+which the time duration that we want to collect needs to be repported along with specific 
+tags.
+
+```go
+func NewMyFilter(cfg baker.InputParams) (baker.Input, error) {
+    return MyFilter{metrics: cfg.Metrics}
+}
+
+func (f *MyFilter) Process(r Record, next func(Record)) {
+ /*... */
+}
+
+func (f *MyFilter) Stats() baker.FilterStats {
+    f.mu.Lock()
+    for _, duration := range f.requestDurations {
+        f.metrics.DurationWithTags(
+            "myfilter_http_request_duration", 
+            duration, 
+            []string{"tag1", "tag2"},
+        )
+    }
+    f.mu.Unlock()
+
+    return baker.FilterStats{
+        NumProcessedLines: atomic.LoadInt64(&f.totalLines),
+        NumFilteredLines: atomic.LoadInt64(&f.filteredLines),
+    }
+}
+```
 
+In the example, we prefered to use the `MetricsClient` in the `Stats` method to limit
+the number of calls to the client. Indeed, calling the `MetricsClient` methods inside 
+the `Process` method could introduce overheads and performance degradations.
```

Comment:
```suggestion
Here, we chose to use the `MetricsClient` in the `Stats` method to limit the number of
calls to the client. Indeed, since `Stats` gets called once per second, the `MetricsClient` is 
also called once per second. We could have directly called `DurationWithTags` from the 
`Process` method but we would have probably - depending on the frequency of incoming 
records - called it way more often, and this could have potentially introduce some performance 
degradation. 
```